ROSCO/src/ControllerBlocks.f90

```diff
@@ -354,6 +356,70 @@ SUBROUTINE SetpointSmoother(LocalVar, CntrPar, objInst)
         ENDIF
 
     END SUBROUTINE SetpointSmoother
+    
+    !-------------------------------------------------------------------------------------------------------------------------------
+    SUBROUTINE LidarConfiguration(LocalVar, CntrPar, objInst)
+    ! Sets the lidar configuration and measurement coordinates
+
+        USE ROSCO_Types, ONLY : LocalVariables, ControlParameters, ObjectInstances
+        IMPLICIT NONE
+    
+        ! Inputs
+        TYPE(ControlParameters),    INTENT(IN   )       :: CntrPar
+        TYPE(LocalVariables),       INTENT(INOUT)       :: LocalVar 
+        TYPE(ObjectInstances),      INTENT(INOUT)       :: objInst
+        
+        REAL(IntKi)                                     :: I        !Loop counter
+      
+              
+   IF (LocalVar%SensorType == 1) THEN   
+        DO I = 1,LocalVar%NumBeam
+            
+         IF (.not. allocated(LocalVar%MsrPosition)) THEN
+             Allocate(LocalVar%MsrPosition(3,5))         
+            LocalVar%MsrPosition(1,I) = LocalVar%MsrPositionsX(I)
```

Comment:
What is MsrPosition?  Please add comments.

Reply:
I will add comments throughout to make it clear what the new variables are.